src/lmdb.ml

```diff
@@ -230,7 +230,7 @@ module Values = struct
     type t
     val default_flags : Flags.t
     val read : db_val -> t
-    val write : t -> db_val
+    val write : (int -> db_val) -> t -> db_val
```

Comment:
To make it really clear what is this function, I would suggest a label : `write: allocate:(int -> db_val) -> t -> db_val`. Otherwise, it's a bit unclear. 

Of course I would like some documentation in the long run, but let's wait for all the changes to land! :p

Reply:
I find the ``write`` function type rather simple and instructive.
Nevertheless it needs elaborate documentation, so the user knows how to use the allocation function and what to return. I will of course provide this documentation.
Given an elaborate documentation I would rather avoid labels in callback functions, because they impose parameter names onto users.

Reply:
Fair enough!